R/dictionary.R

```diff
@@ -71,6 +71,76 @@ reset_rownames <- function(x) {
   x
 }
 
+# Groups -----------------------------------------------------------------
+
+#' Identify groups
+#'
+#' @description
+#'
+#' \Sexpr[results=rd, stage=render]{vctrs:::lifecycle("experimental")}
+#'
+#' * `vec_group_id()` returns an identifier for the group that each element of
+#'   `x` falls in, constructed in the order that they appear. The number of
+#'   groups is also returned as an attribute, `n`.
+#'
+#' * `vec_group_rle()` locates groups in `x` and returns them run length
+#'   encoded in the order that they appear. The return value is a rcrd object
+#'   with fields for the `group` identifiers and the run `length` of the
+#'   corresponding group. The number of groups is also returned as an
+#'   attribute, `n`.
+#'
+#' @param x A vector
+#' @return
```

Comment:
Cross reference to `vec_split_id()`? Or are you imagining that we'll rename that to `vec_group_something()`?

Reply:
`vec_group_pos()`